File: utility_scripts_backup/create_bucket.js
// This is a Node.js script to create a Google Cloud Storage bucket
// You'll need to install the Google Cloud Storage Node.js client library:
// npm install @google-cloud/storage

const {Storage} = require('@google-cloud/storage');

// Path to your service account key file
const keyFilename = './config/credentials/google_service_account.json';

// Create a storage client
const storage = new Storage({
  keyFilename,
  projectId: 'kinetic-harbor-454218-t7'
});

// Create a bucket
const bucketName = 'otakuwarrior-store-bucket';
const location = 'us-central1';  // Choose a location close to your users

async function createBucket() {
  try {
    const [bucket] = await storage.createBucket(bucketName, {
      location,
    });
    console.log(`Bucket ${bucket.name} created.`);
  } catch (error) {
    if (error.code === 409) {
      console.log(`Bucket ${bucketName} already exists.`);
    } else {
      console.error('ERROR:', error);
    }
  }
}

createBucket();

File: utility_scripts_backup/create_bucket.rb
require "google/cloud/storage"

# Path to your service account key file
key_file = "config/credentials/google_service_account.json"

# Create a storage client
storage = Google::Cloud::Storage.new(
  project_id: "kinetic-harbor-454218-t7",
  credentials: key_file
)

# Create a bucket
bucket_name = "otakuwarrior-store-bucket"
location = "us-central1"  # Choose a location close to your users

begin
  bucket = storage.create_bucket bucket_name, location: location
  puts "Bucket #{bucket.name} created."
rescue Google::Cloud::AlreadyExistsError
  puts "Bucket #{bucket_name} already exists."
end